File: LeetCode/11. Container With Most Water/solution.go
func maxArea(height []int) int {
	left := 0
	right := len(height) - 1
	maxA := 0
	for left < right {
		if height[left] < height[right] {
			maxA = max(maxA, (right-left)*height[left])
			left++
		} else {
			maxA = max(maxA, (right-left)*height[right])
			right--
		}
	}
	return maxA
}

File: LeetCode/11. Container With Most Water/solution.py
class Solution:
    def maxArea(self, height: List[int]) -> int:
        res = 0
        left = 0
        right = len(height)-1
        while left < right:
            if height[left] < height[right]:
                res = max(res, (right - left)* height[left])
                left += 1
            else:
                res = max(res, (right - left)* height[right])
                right -= 1
        return res